crates/engarde/src/syntax.rs

```diff
@@ -107,4 +114,24 @@ mod test {
         let output = syntax.format(CODEBLOCK, Some("rust"), Some("base16-ocean.dark"));
         assert_eq!(output, CODEBLOCK_RENDERED.to_string());
     }
+
+    const CUSTOM_CODEBLOCK: &str = "[[[]]]]";
+
+    const CUSTOM_CODEBLOCK_RENDERED: &str = "<pre style=\"background-color:#2b303b;\">\n\
+          <span style=\"color:#c0c5ce;\">[[[]]]</span>\
+          <span style=\"background-color:#bf616a;color:#2b303b;\">]</span>\
+          </pre>\n";
+
+    #[test]
+    fn highlight_custom_syntax() {
+        let mut syntax = Syntax::new();
+        let path = Path::new("./tests/fixtures/");
```

Comment:
We should probably have this in a folder with a little more context as we'll likely add custom syntax support in the future as well.